cpptod.dd

```diff
@@ -829,6 +831,106 @@ void test()
 }
 ------
 
+<hr>$(COMMENT  -------------------------------------------- )
+
+$(H3 <a name="functioninputparameters">Function Input Parameters</a>)
+
+$(H4 The C++ Way)
+
+	$(D C++ does not support explicit input parameters.) If you use a
+	"pass-by-value" parameter `void function(int inputOnlyImplicitly)` it can just
+	be an input parameter implicitly. If you use a "pass-by-reference" parameter
+	usually `const` is applied to the parameter
+	`void function(const int& inputOnlyParameterImplicitly)` that a forbidden
+	modification of the parameter in the function leads to a compilation error.
+
+$(CPPCODE
+void function(int implicitInputParameterPassedByValue) {
+		// some modification dependent on implicitInputParameterPassedByValue
+		...
+}
+
+void function(int& implicitInputParameterPassedByReference) {
+		// some modification dependent on implicitInputParameterPassedByReference
+		...
+}
+)
+
+The functions are used as follows:
+
+$(CPPCODE
+int inputArgumentPassedByValue = 42;
+function(inputArgumentPassedByValue);
+// inputArgumentPassedByValue = 42
+
+int inputArgumentPassedByReference = 42;
+function(const int& inputArgumentPassedByReference);
+// inputArgumentPassedByReference = 42
+)
+
+$(H4 The D Way)
+
+	D supports explicit input parameters.
+
+------
+void function(in int inputParameter) {
+		// some modification dependent on inputParameter
+		...
+}
+------
+
+The function is used as follows:
+
+------
+int inputArgument = 42;
+function(inputArgument);
+// inputArgument = 42
+------
+
+<hr>$(COMMENT  -------------------------------------------- )
+
+$(H3 <a name="functionoutputparameters">Function Output Parameters</a>)
+
+$(H4 The C++ Way)
+
+	$(D C++ does not support explicit output parameters.) You can use a reference
+	parameter to allow the function's parameter serve as input and output
+	to/from the function. (Input only parameters "called by reference" not "called
```

Comment:
- `. (` isn't typographically allowed. Parenthesis can't be a sentence on its own
- I wouldn't quote "called by ...", maybe italic?